projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -277,18 +295,43 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      */
     public show(message?: string): void {
         clearInterval(this.timeoutId);
-        if (message !== undefined) { this.toastMessage = message; }
-        this.onShowing.emit(this);
-        this.isVisible = true;
-        this.animationState = 'visible';
+
+        const overlaySettings: OverlaySettings = {
+            positionStrategy: new GlobalPositionStrategy({
+                horizontalDirection: HorizontalAlignment.Center,
+                verticalDirection:
+                    this.position === 'bottom'
+                        ? VerticalAlignment.Bottom
+                        : this.position === 'middle'
+                        ? VerticalAlignment.Middle
+                        : VerticalAlignment.Top,
+            }),
+            closeOnEscape: false,
+            closeOnOutsideClick: false,
+            modal: false,
+            outlet: this.outlet,
+        };
+
+        if (message !== undefined) {
+            this.toastMessage = message;
+        }
+
+        this.onOpening.subscribe(() => {
```

Comment:
this may leak as on each call to `show`/`hide` a new subscriptions are created. You should subscribe once in `ngOnInit`. Then you should unsubscribe all subscriptions in `ngOnDestroy` with `takeUntil`.
Additionally there is no reason to subscribe to `onShowing` - you may call all the logic you need before you call `super.open()`.